src/useAsyncTasks.ts

```diff
@@ -0,0 +1,184 @@
+import { useEffect, useRef, useReducer, useMemo } from 'react';
+import type { Dispatch, MutableRefObject } from 'react';
+import {
+  asyncWave,
+  type AsyncWaveOptions,
+  PromiseCircularityError,
+} from 'async-wave';
+
+type Task<R> = R | ((input: R) => R | Promise<R> | void | Promise<void>);
+
+type HookOptionProps = {
+  initialLazyDelay: number;
+  successLazyDelay: number;
+};
+
+type HookOptions = {
+  options: Partial<HookOptionProps>;
+};
+
+type SyncOnBefore = { onBefore: () => void };
+
+type Options<R> = Partial<
+  Omit<AsyncWaveOptions<R>, 'onBefore'> & SyncOnBefore & Partial<HookOptions>
+>;
+
+type StateInfo<R> = {
+  isLoading: boolean;
+  data: R | undefined;
+  error: PromiseCircularityError | undefined;
+  isError: boolean;
+  reset: () => void;
+};
+
+const [IS_UNMOUNTED, IS_MOUNTED] = [false, true];
+
+/**
+ * useAsyncTasks
+ * @param {Task<R>[]} tasks - 비동기 작업 리스트
+ * @param {Options<R>} options - 비동기 작업 옵션
+ * @returns {StateInfo<R>} 비동기 작업 상태 정보를 반환
+ */
+const useAsyncTasks = <R>(tasks: Task<R>[], options: Options<R>) => {
+  const isMountedRef = useRef(IS_UNMOUNTED); // 컴포넌트가 언마운트된 후 비동기 작업이 완료될 때 상태 업데이트를 방지
+  const [state, dispatch] = useReducer(reducer<R>, {
+    isLoading: false,
+    data: undefined,
+    error: undefined,
+  });
+
+  useEffect(() => {
+    isMountedRef.current = IS_MOUNTED;
+
+    (async () => {
+      await asyncWave<R>(
+        isMountedRef.current ? [...tasks] : [],
+        generateTaskHandlers(dispatch, isMountedRef, { ...options })
+      );
+    })();
+
+    return () => {
+      isMountedRef.current = IS_UNMOUNTED;
+    };
+  }, []);
+
+  const stateInfo: StateInfo<R> = useMemo(
+    () => ({
+      isLoading: state.isLoading,
+      data: state.data,
+      error: state.error,
+      isError: Boolean(state.error),
+      reset: () => {
```

Comment:
```suggestion
      reset() {
```

[P1] 코딩컨벤션 > Object > 메서드의 단축 구문 부분 확인 부탁드립니다!

Reply:
앗 반영하겠습니다 😅

Reply:
[commit] https://github.com/frontend-opensource-project/use-react-hooks/pull/3/commits/4c41e9de099fa818b0daf99df45b86188f6216de

@bicochan  반영했습니다!